Watchman.DomainModel/Settings/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,90 @@
+        {
+            this._sessionFactory = sessionFactory;
+            this._configurationMapperService = configurationMapperService;
+            this._configurationTypesSearcher = configurationTypesSearcher;
+        }
+
+        public T GetConfigurationItem<T>(ulong serverId) where T : IMappedConfiguration
+        {
+            if (this._lastRefreshed < DateTime.Now.AddMinutes(-10))
+            {
+                this.ReloadConfiguration();
+            }
+            var configurations = this._cachedConfigurationItem[typeof(T)];
+            var serverConfiguration = configurations.GetValueOrDefault(serverId) ?? configurations[0];
+            return (T)serverConfiguration;
+        }
+
+        public IEnumerable<IMappedConfiguration> GetConfigurationItems(ulong serverId)
+        {
+            if (this._lastRefreshed < DateTime.Now.AddMinutes(-10))
+            {
+                this.ReloadConfiguration();
+            }
+            return this._cachedConfigurationItem.Select(x => x.Value.GetValueOrDefault(serverId) ?? x.Value[0]);
+        }
+
+        public async Task SaveNewConfiguration(IMappedConfiguration changedConfiguration)
+        {
+            using var session = this._sessionFactory.Create();
+            var existingConfiguration = session.Get<ConfigurationItem>()
+                .FirstOrDefault(x => x.ServerId == changedConfiguration.ServerId && x.Name == changedConfiguration.Name);
+            var baseFormatConfigurationItem = this._configurationMapperService.MapIntoBaseFormat(changedConfiguration);
+
+            if (existingConfiguration == null)
+            {
+                await session.AddAsync(baseFormatConfigurationItem);
+            }
+            else
+            {
+                existingConfiguration.SetValue(baseFormatConfigurationItem.Value);
+                await session.AddOrUpdateAsync(existingConfiguration);
+            }
+        }
+
+        public async Task InitDefaultConfigurations()
+        {
+            var configurationsTypes = this._configurationTypesSearcher.ConfigurationTypes;
+            var configurations = configurationsTypes.Select(x =>
+            {
+                var conf = (IMappedConfiguration)Activator.CreateInstance(x);
+                return this._configurationMapperService.MapIntoBaseFormat(conf);
+            });
+            using (var session = this._sessionFactory.Create())
+            {
+                var existingConfigurations = session.Get<ConfigurationItem>().Where(x => x.ServerId == 0).ToList();
+
+                foreach (var configuration in configurations.Where(x => existingConfigurations.All(y => x.Name != y.Name)))
+                {
+                    await session.AddAsync(configuration);
+                }
+            }
+            this.ReloadConfiguration();
+        }
+
+        private void ReloadConfiguration()
+        {
```

Comment:
tutaj mógłbyś przekazywać sesje

Reply:
chociaż w sumie

Reply:
w sumie

Reply:
przekazuj ISession session = null
w środku sprawdzenie czy jest nullem
jak jest nullem to `using session = this._sessionFactory.Create();`